packages/@aws-cdk/aws-servicediscovery/lib/namespace.ts

```diff
@@ -0,0 +1,145 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import cdk = require('@aws-cdk/cdk');
+import { CfnHttpNamespace, CfnPrivateDnsNamespace, CfnPublicDnsNamespace} from './servicediscovery.generated';
+
+export interface INamespace extends cdk.IConstruct {
+  /**
+   * A name for the Namespace.
+   */
+  readonly name: string;
```

Comment:
I find `name` a little generic. I'd prefer `namespaceName` to be consistent with the other attributes.